ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -53,1019 +57,618 @@ namespace ng = ngraph;
-            "tf_function_error_" + ctx->op_kernel().name() + ".json",
-            ng_function);
-        return errors::Internal("Caught exception while compiling op_backend: ",
-                                exp.what(), "\n");
-      } catch (...) {
-        BackendManager::UnlockBackend(m_op_backend_name);
-        NgraphSerialize(
-            "tf_function_error_" + ctx->op_kernel().name() + ".json",
-            ng_function);
-        return errors::Internal("Error in compiling op_backend\n");
-      }
-      BackendManager::UnlockBackend(m_op_backend_name);
-      event_compile.Stop();
-      ngraph::Event::write_trace(event_compile);
-
-      m_ng_exec_map[signature] = ng_exec;
-      // caching ng_function to serialize to ngraph if needed
-      m_ng_function_map[ng_exec] = ng_function;
-
-      m_lru.push_front(signature);
-      // Memory after
-      MemoryProfile(vm, rss);
-      auto delta_vm_mem = vm - vm0;
-      auto delta_res_mem = rss - rss0;
-      NGRAPH_VLOG(1) << "NGRAPH_TF_CACHE_PROFILE: OP_ID: " << my_instance_id
-                     << " Step_ID: " << ctx->step_id()
-                     << " Cache length: " << m_ng_exec_map.size()
-                     << "  Cluster: " << ctx->op_kernel().name()
-                     << " Delta VM: " << delta_vm_mem
-                     << "  Delta RSS: " << delta_res_mem
-                     << "  Function size: " << function_size
-                     << " KB Total RSS: " << rss / (1024 * 1024) << " GB "
-                     << " VM: " << vm / (1024 * 1024) << " GB" << endl;
-    }  // end of input signature not found in m_ng_exec_map
-    else {
-      // Found the input signature in m_ng_exec_map, use the cached executable
-      // Update the m_lru
-      if (signature != m_lru.front()) {
-        m_lru.remove(signature);
-        m_lru.push_front(signature);
-      }
-      ng_exec = it->second;
+    // Create the TF output tensor
+    vector<int64> dims;
+    for (auto dim : ng_shape) {
+      dims.push_back(dim);
     }
-    return Status::OK();
+    TensorShape tf_shape(dims);
+    Tensor* output_tensor = nullptr;
+    OP_REQUIRES_OK(ctx, ctx->allocate_output(i, tf_shape, &output_tensor));
+    tf_output_tensors.push_back(output_tensor);
+
+    // Make sure the nGraph-inferred element type agrees with what TensorFlow
+    // expected.
+    ng::element::Type expected_elem_type;
+    OP_REQUIRES_OK(ctx,
+                   TFDataTypeToNGraphElementType(ctx->expected_output_dtype(i),
+                                                 &expected_elem_type));
+    OP_REQUIRES(
+        ctx, ng_element_type == expected_elem_type,
+        errors::Internal("Element type inferred by nGraph does not match "
+                         "the element type expected by TensorFlow"));
+  }
+// if the output tensor is going to be assigned to a variable
+// we ask nGraph to provide the output directly in the variable tensor
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
```

Comment:
This #ifdefined needs to be inside the loop for NG outputs creation. Needs to be moved inside 
AllocateNGOutputTensors